src/wrap/mod.rs

```diff
@@ -42,33 +42,56 @@ impl Checker for Wrapper {
             SuggestionSet::new(),
             |mut acc, (origin, chunks)| {
                 for chunk in chunks {
-                    let mut new_lines = chunk
+                    let mut new_lines = dbg!(&chunk)
                         .as_str()
                         .split("\n\n")
                         .collect::<Vec<&str>>()
                         .iter()
-                        .map(|s| s.replace("\n", "") )
+                        .map(|s| s.replace("\n", ""))
                         .fold::<Vec<String>, _>(Vec::new(), |mut acc, comment| {
                             let mut new_comment = wrapper
                                 .wrap_iter(comment.trim())
-                                .map(|b| b.into_owned()).collect();
+                                .map(|b| b.into_owned())
+                                .collect();
                             acc.append(&mut new_comment);
                             acc.push("".into());
                             acc
                         });
                     // remove last newline
                     let _ = new_lines.pop();
+                    // @todo that's too easy
+                    if new_lines.len() == chunk.as_str().lines().count() {
+                        log::trace!("No rewrapping required for '{:?}'", chunk);
+                        break; // the chunk did not change
+                    }
                     // @todo find proper span and range
-                    let range = 0..chunk.as_str().len();
-                    let span = Span { start: LineColumn { line: 0, column: 0}, end: LineColumn { line: 1, column: 5}};
+                    let range = dbg!(0..chunk.as_str().len());
+                    let mut start = 1000..1001;
+                    let mut end = 0..1;
```

Comment:
avoid initializing with garbage values - try to use the first element of the list `.iter().next()`, and adjust the remaining `.iter().skip(1)` or use a `mut it = chunk.iter(); let first = iter.next(); for each in iter {... }`

Reply:
I tried to avoid that, but did not think of `iter().next()` and friends :) I didn't really like the whole loop impl at all..